File: JAVA/Arrange Consonants and Vowels.java
class Solution {
     public Node arrangeCV(Node head){
        if(head == null) return head;
        Node temp = head;
        Node vowels = new Node('a');
        Node vhead = vowels;
        while(temp.next != null){
            while(temp.next != null && temp.next.data != 'a' && temp.next.data != 'e' && temp.next.data != 'i' && temp.next.data != 'o' && temp.next.data != 'u' ){
                temp = temp.next;
            }
            if(temp.next != null){
                vowels.next = temp.next;
                vowels = vowels.next;
                if(temp.next.next == null) 
                    temp.next = null;
                else
                    temp.next = temp.next.next;
            }
        }
        vowels.next = head;
        if(head.data == 'a' || head.data == 'e' || head.data == 'i' || head.data == 'o' || head.data == 'u'){
            vowels.next = head.next;
            vhead = vhead.next;
            head.next = vhead;
            return head;
        }
        return vhead.next;
    }
}

File: C++/Arrange Consonants and Vowels.cpp
class Solution {
  public:
     struct Node* arrangeCV(Node* head) 
    {
        Node* consonantHead = nullptr;
        Node* consonantTail = nullptr;
        Node* vowelsHead = nullptr;
        Node* vowelsTail = nullptr;
        Node* temp = head;
        while(temp != nullptr)
        {
            char c = temp ->data;
            if(c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u')
            {
                Node* newnode = new Node(c);
                if(vowelsHead == nullptr)
                {
                    vowelsHead = vowelsTail = newnode;
                }
                else
                {
                    vowelsTail ->next = newnode;
                    vowelsTail = vowelsTail ->next;
                }
            }
            else
            {
                Node* newnode = new Node(c);
                if(consonantHead == nullptr)
                {
                    consonantHead = consonantTail = newnode;
                }
                else
                {
                    consonantTail ->next = newnode;
                    consonantTail = consonantTail ->next;
                }
            }
            temp = temp->next;
        }
        if(vowelsHead == nullptr)
        {
            return consonantHead;
        }
        vowelsTail ->next = consonantHead;
        return vowelsHead;
    }
};
